### PR TITLE
Fix block explorer links

### LINKED ISSUE #338: Fix block explorer links
**Description**
Clicking links of address is inconsistent across the app. We should have one component that takes users to block explorers relevant page 

**Steps to Reproduce**

1. Go to proposa
2. Expand action
3. Click address with link icon

Also check other parts of app with link icon

### PR DESCRIPTION
**Pull request by rulfo71**
# Description

Fixing block explorer links through the app 

Closes #338 

image
image

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Manually for now

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any UI changes have been tested and made responsive for mobile views

### CODE REVIEW

**Comment by Filipv95** [comment]
Hey @rulfo71, is this change applying only on addresses withing expanded Proposal Action on Proposal page?

**Comment by rulfo71** [comment]
hey @Filipv95 ! 
for now it's done only for the one you mentioned, on create proposal page but when i finish it it will be for all addresses

**Comment by rulfo71** [comment]
at the moment is finished missing tests but missing tests on component BlockExplorerLink

File: src/components/ActionsBuilder/SupportedActions/UpdateENSContent/UpdateENSContentSummary.tsx [hunk 1]
```diff
@@ -89,28 +84,18 @@ const UpdateENSContentSummary: React.FC<ActionViewProps> = ({
       <DetailHeader>{t('ens.domain')}</DetailHeader>
       <DetailRow>
         <DetailBody>
-          <StyledSegmentLink
-            href={blockExplorerUrl}
-            target="_blank"
-            rel="noopener"
-          >
+          <ExternalLink href={blockExplorerUrl}>
```
Comment (dcrescimbeni) [inline src/components/ActionsBuilder/SupportedActions/UpdateENSContent/UpdateENSContentSummary.tsx:87]: I think we should use `<BlockExplorerLink />` component here

File: src/components/primitives/Links/ExternalLink.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+export const ExternalLink: React.FC<ExternalLinkProps> = ({
+  href,
+  children,
+  unstyled,
+}) => {
+  if (unstyled) {
+    return (
+      <UnstyledExternalLink href={href} target="_blank" rel="noopener">
+        <ParamDetail>
```
Comment (dcrescimbeni) [inline src/components/primitives/Links/ExternalLink.tsx:16]: The styled and unstyled links place the icon differently, caused by the `<ParamDetail />` component.

I think the behaviour in the unstyled component is the desired one (the icon besides the text), but I'd add some spacing between the elements.

image

File: src/components/primitives/Links/ExternalLink.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { ParamDetail } from 'components/ActionsBuilder/CallDetails/CallDetails.styled';
+import { BiLinkExternal } from 'react-icons/bi';
+import { FiExternalLink } from 'react-icons/fi';
```
[the remark this answers is not in the document]
Comment (rulfo71) [reply src/components/primitives/Links/ExternalLink.tsx]: i just removed the unstyle one, didn't make much sense

File: src/components/primitives/Links/styles.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  overflow-wrap: break-word;
+  &:hover {
+    cursor: pointer;
+    color: ${({ theme }) => theme.colors.text};
+  }
+`;
+
+export const FlexContainer = styled.div`
+  display: flex;
+`;
```
Comment (dcrescimbeni) [inline src/components/primitives/Links/styles.tsx:20]:
```suggestion
export const FlexContainer = styled.div`
  display: flex;
`;

```

Removing this (see comment above)

File: src/components/primitives/Links/ExternalLink.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      </StyledSegmentLink>
+    );
+  }
+
+  return (
+    <StyledSegmentLink href={href} target="_blank" rel="noopener">
+      <LinkDetail>{children}</LinkDetail>
+      <FiExternalLink />
+    </StyledSegmentLink>
+  );
```
Comment (dcrescimbeni) [inline src/components/primitives/Links/ExternalLink.tsx:25]: I think this should be only one component and handle the styling logic in the styled component

Comment (rulfo71) [reply src/components/primitives/Links/ExternalLink.tsx:25]: thanks! yes i didn't realize this is not needed anymore.

File: src/components/primitives/Links/styles.tsx (hunk 4 above)
Comment (rulfo71) [reply src/components/primitives/Links/styles.tsx:20]: great to know

File: src/components/primitives/Links/BlockExplorerLink.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  avatarSize = 24,
+}) => {
+  const { chain } = useNetwork();
+  const { ensName, imageUrl } = useENSAvatar(address, MAINNET_ID);
+  if (!address) return null;
+
+  const blockExplorerUrl = getBlockExplorerUrl(chain, address, 'address');
+
+  return (
+    <FlexContainer>
```
[the remark this answers is not in the document]
Comment (rulfo71) [reply src/components/primitives/Links/BlockExplorerLink.tsx:35]: great to know, i meant here lol

**Review by dcrescimbeni — APPROVED** [review]

**Review by dcrescimbeni — DISMISSED** [review]

File: src/components/primitives/Links/BlockExplorerLink.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
+import { Segment } from 'components/ActionsBuilder/SupportedActions/common/infoLine';
+import { Avatar } from 'components/Avatar';
+import useENSAvatar from 'hooks/Guilds/ens/useENSAvatar';
+import { MAINNET_ID, shortenAddress } from 'utils';
+import { Chain, useNetwork } from 'wagmi';
+import { Flex } from '../Layout';
+import { ExternalLink } from './ExternalLink';
+// import { FlexContainer } from './styles';
```
Comment (Kenny-Gin1) [inline src/components/primitives/Links/BlockExplorerLink.tsx:8]: Remove comments

File: src/components/primitives/Links/BlockExplorerLink.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+import { Avatar } from 'components/Avatar';
+import useENSAvatar from 'hooks/Guilds/ens/useENSAvatar';
+import { MAINNET_ID, shortenAddress } from 'utils';
+import { Chain, useNetwork } from 'wagmi';
+import { Flex } from '../Layout';
+import { ExternalLink } from './ExternalLink';
+// import { FlexContainer } from './styles';
+import { BlockExplorerLinkProps } from './types';
+
+// this code is duplicated from provider/chain.ts because there was a circular dependency when testing
```
Comment (Kenny-Gin1) [inline src/components/primitives/Links/BlockExplorerLink.tsx:11]: Is there an action point? Or can this comment be removed?

**Review by Kenny-Gin1 — APPROVED** [review]
Clean! 👍

File: src/components/primitives/Links/ExternalLink.tsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import {} from 'components/ActionsBuilder/CallDetails/CallDetails.styled';
```
Comment (MiltonTulli) [inline src/components/primitives/Links/ExternalLink.tsx:1]:
```suggestion

```

**Comment by rossneilson** [comment]
Could we also change blockscout to link to gnosis scan https://gnosisscan.io/

**Review by rossneilson — APPROVED** [review]

**Pull request merged**